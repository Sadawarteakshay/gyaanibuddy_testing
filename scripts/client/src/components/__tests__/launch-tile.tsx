import React from 'react';

import { render, cleanup } from '../../test-utils';
import LaunchTile from '../launch-tile';
import { renderApollo } from '../../test-utils';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('Launch Tile', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('renders without error', () => {
    renderApollo(
      <LaunchTile
        launch={{
          __typename: 'Launch',
          isBooked: false,
          id: '1',
          mission: { name: 'the first one', __typename: 'Mission', missionPatch: null },
          rocket: { name: 'harambe', __typename: 'Rocket', id: '1' },
        }}
      />,
    );
  });

  it('render snapshot', () => {
    const LaunchTileElement = <LaunchTile
      launch={{
        __typename: 'Launch',
        isBooked: false,
        id: '1',
        mission: { name: 'the first one', __typename: 'Mission', missionPatch: null },
        rocket: { name: 'harambe', __typename: 'Rocket', id: '1' },
      }}
    />;

    // see https://stackoverflow.com/questions/70805929/how-to-fix-error-usehref-may-be-used-only-in-the-context-of-a-router-compon
    render(LaunchTileElement, {wrapper: MemoryRouter});
    
    // Snapshot test - ensure button has no unsuspecting changes between commits
    const ElementTree = renderer
      .create(<MemoryRouter>{LaunchTileElement}</MemoryRouter>)
      .toJSON();

    expect(ElementTree).toMatchSnapshot();
  });
});
