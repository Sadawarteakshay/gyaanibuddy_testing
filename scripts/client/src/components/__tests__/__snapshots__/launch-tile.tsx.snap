// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Tile render snapshot 1`] = `
<a
  className="css-s37lsq"
  href="/launch/1"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(iss.jpg)",
    }
  }
>
  <h3>
    the first one
  </h3>
  <h5>
    harambe
  </h5>
</a>
`;
